internal/engine/eval/rego/datasources.go

```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: Copyright 2023 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package rego
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/types"
+
+	v1datasources "github.com/mindersec/minder/pkg/datasources/v1"
+)
+
+// RegisterDataSource implements the Eval interface.
+func (e *Evaluator) RegisterDataSource(ds v1datasources.DataSource) {
+	for key, dsf := range ds.GetFuncs() {
+		fmt.Printf("Registering data source %s\n", key)
+		e.regoOpts = append(e.regoOpts, buildFromDataSource(key, dsf))
```

Comment:
**question:** I believe there might be a potential issue here that we have to address: we need to make sure functions are in the "namespace" of the Data Source they're declared in, and that they do not shadow other identifiers in REGO.

As an example, assume I have two Data Sources, `foo` and `bar`, and I want to use them both in a single Rule; additionally, let's assume both Data Sources expose a function named  `garble` with the same signature; I would expect to be able to use both `foo.garble(...)` and `bar.garble(...)`.

Where is the final name defined? My understanding is that it would be the value of `key`, am I right? If that's the case, we need to either document that the name MUST contain the data source name, or we must turn registration inside-out.

Proposal: assuming `key` is the final name, Minder must be in control of it, and not the final User, and we could just change the interface to return a collection of `func(_ rego.BuiltinContext, obj *ast.Term) (*ast.Term, error)` and let the upper layer figure out the final name.

Reply:
Solved in `func (reg *DataSourceRegistry) RegisterDataSource(name string, ds DataSource) (err error)`